internal/modulecontext/module_context.go

```diff
@@ -175,11 +196,34 @@ func (m ModuleContext) BehaviorForVerb(ref schema.Ref) (optional.Option[VerbBeha
 	return optional.None[VerbBehavior](), nil
 }
 
+// BehaviorForMap returns what to do to execute a map
+//
+// This allows module context to dictate behavior based on testing options
+// Returning optional.Nil indicates the map should be executed normally via the controller
+func (m ModuleContext) BehaviorForMap(ref schema.Ref) (optional.Option[MapBehavior], error) {
```

Comment:
I don't think this should be in the `ModuleContext` should it? The behaviour should be set on the `fakeFTL` implementation.

It's a bit confusing because we've not migrated to the `FTL` interface fully.

Reply:
Ahhhh that explains a lot, thanks! I'll shift it over and that should make things look a lot saner